x-pack/plugins/index_management/server/routes/api/inference_models/register_create_elasticsearch_inference_route.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { addBasePath } from '..';
+import { RouteDependencies } from '../../../types';
+
+export function registerCreateElasticsearchInferenceRoute({
+  router,
+  lib: { handleEsError },
+  config,
+}: RouteDependencies) {
+  const paramsSchema = schema.object({
+    inferenceId: schema.string(),
+  });
+  const bodySchema = schema.object({
+    trainedModelId: schema.string(),
+  });
+  // Create text_embedding inference
+  router.post(
+    {
+      path: addBasePath('/inference/text_embedding/{inferenceId}'),
+      validate: { body: bodySchema, params: paramsSchema },
```

Comment:
I think this overlaps with the create Inference API route that's added in [inference flyout PR](https://github.com/elastic/kibana/blob/49b2c11f24e12da9f19e528f9ee4fac7fd8d2ce5/x-pack/plugins/index_management/server/routes/api/inference_models/register_put_inference_route.ts) 

We probably need to decide which one to keep.

Reply:
I agree that we should utilize either one of them and discard other one.